libsolidity/interface/StandardCompiler.cpp

```diff
@@ -337,7 +337,7 @@ Json::Value StandardCompiler::compileInternal(Json::Value const& _input)
 	}
 
 	Json::Value contractsOutput = Json::objectValue;
-	for (string const& contractName: m_compilerStack.contractNames())
+	for (string const& contractName: success ? m_compilerStack.contractNames() : vector<string>())
```

Comment:
The problem here is that previously, we aborted on `!success` much earlier. Now `sourceNames()` above is valid if `!success`, but `contractNames()` throws and the Why3 (following this) also operates on `sourceNames()`.